compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -891,14 +891,14 @@ internal fun KSAnnotated.findAnnotationFromUseSiteTarget(): Collection<KSAnnotat
             }
             val annotationsFromParents = mutableListOf<KSAnnotation>()
             (parent as? KSPropertyAccessorImpl)?.let {
-                annotationsFromParents.addAll(it.originalAnnotations.filter { it.useSiteTarget == AnnotationUseSiteTarget.SETPARAM })
+                annotationsFromParents.addAll(it.originalAnnotations.asSequence().filter { it.useSiteTarget == AnnotationUseSiteTarget.SETPARAM })
                 parent = (parent as KSPropertyAccessorImpl).receiver
             }
             (parent as? KSPropertyDeclarationImpl)?.let {
-                annotationsFromParents.addAll(it.originalAnnotations.filter { it.useSiteTarget == AnnotationUseSiteTarget.SETPARAM })
+                annotationsFromParents.addAll(it.originalAnnotations.asSequence().filter { it.useSiteTarget == AnnotationUseSiteTarget.SETPARAM })
             }
-            annotationsFromParents
+            annotationsFromParents.asSequence()
```

Comment:
// TODO: eliminate `annotationsFromParents` and compute the Sequence on the fly.